navis/nbl/smat.py

```diff
@@ -54,88 +54,104 @@ def yield_not_same(pairs: Iterable[Tuple[Any, Any]]) -> Iterator[Tuple[Any, Any]
             yield a, b
 
 
-def concat_results(results: Iterable[List[np.ndarray]]) -> List[np.ndarray]:
+def concat_results(results: Iterable[List[np.ndarray]],
+                   total: Optional[int] = None,
+                   desc: str = 'Querying',
+                   progress: bool = True) -> List[np.ndarray]:
+    """Helper function to concatenate batches of e.g. [(dist, dots), (dist, dots)]
+    into single (dist, dot) arrays.
+    """
     intermediate = defaultdict(list)
-    for result_lst in results:
-        for idx, array in enumerate(result_lst):
-            intermediate[idx].append(array)
+    with config.tqdm(desc=desc,
+                     total=total,
+                     leave=False,
+                     disable=not progress) as pbar:
+        for result_lst in results:
+            for idx, array in enumerate(result_lst):
+                intermediate[idx].append(array)
+            pbar.update(1)
 
     return [np.concatenate(arrs) for arrs in intermediate.values()]
 
 def _nblast_v1_scoring(dist : float, dp : float, sigma_scoring : int = 10):
-    """NBLAST analytical scoring function following Kohl et al. (2013)
+    """NBLAST analytical scoring function following Kohl et al. (2013).
 
     Parameters
     ----------
-    dist :          float
+    dist :          float | array thereof
                     Distance between two points.
-    dp :            float
+    dp :            float | array thereof
                     Absolute dot product between points.
     sigma_scoring : int
                     Sigma of the exponential decrease.
-                    It determines how close in space points must be to be considered similar
-                    Defaults to 10.
+                    It determines how close in space points must be to be
+                    considered similar. Defaults to 10.
     Returns
     -------
-    scores :        float
-                    Score value.
+    scores :        float or array thereof
+                    Score value(s).
 
     References
     ----------
     Kohl J, Ostrovsky AD, Frechter S, Jefferis GS. A bidirectional circuit switch
-    reroutes pheromone signals in male and female brains. Cell. 2013 Dec;155(7) 1610-1623.
-    doi: 10.1016/j.cell.2013.11.025. 
+    reroutes pheromone signals in male and female brains.
+    Cell. 2013 Dec;155(7) 1610-1623. doi: 10.1016/j.cell.2013.11.025.
     """
     return np.sqrt(np.abs(dp) * np.exp(-(dist ** 2)/(2 * sigma_scoring ** 2)))
 
 
-DotpropKey = Hashable
+NeuronKey = Hashable
 
 
 class LookupNdBuilder:
     def __init__(
         self,
-        dotprops: Union[List[Dotprops], Mapping[DotpropKey, Dotprops]],
-        matching_lists: List[List[DotpropKey]],
-        match_fn: Callable[[Dotprops, Dotprops], List[np.ndarray]],
-        nonmatching_list: Optional[List[DotpropKey]] = None,
+        neurons: Union[List[object], Mapping[NeuronKey, object]],
```

Comment:
A few changes in the docs and naming of variables (here and below) to make it clear this is not dot props-specific but can be very generic.

Reply:
Good call. I'll make those changes!